src/components/tooltip/TooltipDirective.js

```diff
@@ -118,7 +118,10 @@ goog.require('ga_topic_service');
                 updateCursorStyle, 10, false, false);
 
         // Register click/touch/mousemove events on map
-        var registerMapEvents = function(scope, onClick) {
+        var registerMapEvents = function(scope, onClick, deregEvent) {
```

Comment:
Something like

```
var registerMapEvents = function() {
    var dereg = angular.noop;
    return function(scope, onClick) {
        if (dereg != angular.noop) {
            return dereg;
        }
        .
        .
        .
        dereg = function() {
            deregMapClick();
            ol.Observable.unByKey(deregPointerMove);
            dereg = angular.noop;
        }
        return dereg;
    };
}();
```

That way, you don't need to track results of dereg calls.
Note: code above not tested.